ui/src/dashboards/components/Dashboard.tsx

```diff
@@ -46,24 +47,34 @@ class DashboardComponent extends PureComponent<Props> {
       manualRefresh,
       onPositionChange,
       onAddCell,
+      viewsStatus,
     } = this.props
 
     return (
       <SpinnerContainer loading={status} spinnerComponent={<TechnoSpinner />}>
-        <Page.Contents fullWidth={true} scrollable={true} className="dashboard">
-          {!!cells.length ? (
-            <Cells
-              cells={cells}
-              timeRange={timeRange}
-              manualRefresh={manualRefresh}
-              onPositionChange={onPositionChange}
-            />
-          ) : (
-            <DashboardEmpty onAddCell={onAddCell} />
-          )}
-          {/* This element is used as a portal container for note tooltips in cell headers */}
-          <div className="cell-header-note-tooltip-container" />
-        </Page.Contents>
+        <SpinnerContainer
+          loading={viewsStatus}
+          spinnerComponent={<TechnoSpinner />}
+        >
+          <Page.Contents
+            fullWidth={true}
+            scrollable={true}
+            className="dashboard"
+          >
+            {!!cells.length ? (
```

Comment:
perhaps this should check whether views have loaded? cells array sometimes comes back as [undefined, undefined...]